src/api/entities/Asset/BaseAsset/index.ts

```diff
@@ -122,4 +136,99 @@ export abstract class BaseAsset extends Entity<UniqueIdentifiers, string> {
 
     return boolToBoolean(result);
   }
+
+  /**
+   * Retrieve the Asset's data
+   *
+   * @note can be subscribed to
+   */
+  protected internalDetails(): Promise<AssetDetails>;
+  protected internalDetails(callback: SubCallback<AssetDetails>): Promise<UnsubCallback>;
```

Comment:
At the time of writing I thought I wasn't able to access `super` in `NftCollection`, but it seems like that is possible after all. Going to see if I can leave this as just `details`